src/django/api/permissions.py

```diff
@@ -0,0 +1,27 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+from .models.user import Roles
+
+
+def request_is_read_only(request):
+    return request.method in SAFE_METHODS
+
+
+class UserCanReviewBoundaries(BasePermission):
```

Comment:
Nice work! I've been thinking about doing this for a while.

Consider using `CanReview` and `CanWrite` to match the permissions in the front-end.